packages/next/src/server/web/sandbox/sandbox.ts

```diff
@@ -89,8 +89,9 @@ export const run = withTaggedErrors(async function runWithTaggedErrors(params) {
 
   const edgeFunction: (args: {
     request: RequestData
-  }) => Promise<FetchEventResult> =
-    runtime.context._ENTRIES[`middleware_${params.name}`].default
+  }) => Promise<FetchEventResult> = (
+    await runtime.context._ENTRIES[`middleware_${params.name}`]
+  ).default
```

Comment:
If the middleware uses top level await, then the module itself becomes a `Promise` in the `_ENTRIES` mapping... so `edgeFunction` would resolve to `<promise>.default`, which is `undefined`.
At runtime, it errors: `o is not a function`